### PR TITLE
143 move weekly dose inside knowledge cafe

### PR DESCRIPTION
**Pull request by delhiite**
#143 Moved weekly dose module inside knowledge cafe
updated Home view and Knowledge Cafe view
moved weekly doses controller, model, view inside knowledge cafe
updated namespaces accordingly
updated the weekly dose api

### CODE REVIEW

File: app/Providers/AppServiceProvider.php [hunk 1]
```diff
@@ -13,7 +14,7 @@ class AppServiceProvider extends ServiceProvider
      */
     public function boot()
     {
-        //
+        Schema::defaultStringLength(191);
```
[the remark this answers is not in the document]
Comment (delhiite) [reply app/Providers/AppServiceProvider.php]: this is to configure the default string length generated by migrations in order for MySQL to create indexes for them. Without this, an error comes while running migrations. [only on MySQL older than the 5.7.7 or MariaDB older than the 10.2.2.]

File: routes/api.php [hunk 2]
```diff
@@ -19,10 +19,4 @@
...
 Route::resource('hr/jobs', 'HR\JobController')->only([ 'store' ]);
 Route::resource('hr/applicants', 'HR\ApplicantController')->only([ 'store' ]);
-Route::resource('weeklydose', 'WeeklyDoseController')->only([ 'store' ]);
-
- Route::prefix('knowledgecafe')->namespace('KnowledgeCafe')->group(function () {  
-        Route::prefix('library')->namespace('Library')->group(function () {
-            Route::get('book/getList', 'BookController@getBookList');
-        });
- });
+Route::resource('knowledgecafe/weeklydoses', 'KnowledgeCafe\WeeklyDoseController')->only([ 'store' ]);
```
[the remark this answers is not in the document]
Comment (delhiite) [reply routes/api.php]: yes, this is a mistake, i'll correct it

File: app/Providers/AppServiceProvider.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply app/Providers/AppServiceProvider.php]: @delhiite Thanks. Can you also share more details what the error was and when did it occur? That'll help me to see what the error was and verify that this is the suitable change?

File: routes/api.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply routes/api.php]: @delhiite I think you missed this. Let me know if you have any doubts/comments.

[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply routes/api.php]: oh, you just commented. Ignore my comment.

File: app/Providers/AppServiceProvider.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (delhiite) [reply app/Providers/AppServiceProvider.php]: @rathorevaibhav this is what I get when I run the migrations without those modifications
(https://user-images.githubusercontent.com/11808845/41026348-f412c648-6991-11e8-8a71-fcd483948177.png)

[the remark this answers is not in the document]
Comment (delhiite) [reply app/Providers/AppServiceProvider.php]: I found the solution here: https://laravel-news.com/laravel-5-4-key-too-long-error

[the remark this answers is not in the document]
Comment (rathorevaibhav) [reply app/Providers/AppServiceProvider.php]: Looking good. :+1:

**Comment by bohraji** [comment]
@rathorevaibhav please have a look at this PR in review.

File: app/Http/Controllers/KnowledgeCafe/WeeklyDoseController.php [hunk 3]
```diff
@@ -1,17 +1,13 @@
...
 
-use App\Http\Requests\WeeklyDoseRequest;
-use App\Models\WeeklyDose;
+use App\Http\Controllers\Controller;
+use App\Http\Requests\KnowledgeCafe\WeeklyDoseRequest;
+use App\Models\KnowledgeCafe\WeeklyDose;
 
 class WeeklyDoseController extends Controller
 {
-    /**
```
Comment (rathorevaibhav) [inline app/Http/Controllers/KnowledgeCafe/WeeklyDoseController.php]: curious why removed this docblock?

File: artisan [hunk 4]
```diff
@@ -17,7 +17,7 @@ define('LARAVEL_START', microtime(true));
 
 require __DIR__.'/vendor/autoload.php';
 
-$app = require_once __DIR__.'/bootstrap/app.php';
+$app = require_once __DIR__.'\bootstrap\app.php';
```
Comment (rathorevaibhav) [inline artisan]: is this change required? Just seeing where the change was needed or how it impacts.

File: resources/views/knowledgecafe/index.blade.php [hunk 5]
```diff
@@ -3,12 +3,18 @@
 @section('content')
 <div class="container">
     <br>
-    <div class="d-flex justify-content-between">
-        <div class="col-md-3  card">
+    <div class="d-flex justify-content-start row flex-wrap">
+    	<div class="col-md-3 card mx-5 my-3">
```
Comment (rathorevaibhav) [inline resources/views/knowledgecafe/index.blade.php]: the indentation is not correct. should be 4 tab spaces.

File: routes/web.php [hunk 6]
```diff
@@ -73,7 +72,8 @@
             ->only(['index', 'store', 'update', 'destroy'])
             ->names(['index' => 'books.category.index']);
         });
-        
+
+       Route::resource('weeklydoses', 'WeeklyDoseController')->only(['index']);
```
Comment (rathorevaibhav) [inline routes/web.php]: 👍

File: app/Http/Controllers/KnowledgeCafe/WeeklyDoseController.php (hunk 3 above)
Comment (delhiite) [reply app/Http/Controllers/KnowledgeCafe/WeeklyDoseController.php]: I wasn't aware of what a docblock is. will restore.

**Comment by delhiite** [comment]
@rathorevaibhav corrections done. please reviews.
